core/controllers/access_validators_test.py

```diff
@@ -416,3 +418,59 @@ def test_validation_returns_false_if_given_user_is_non_existent(
             ), expected_status_int=404
         )
         self.logout()
+
+
+class TopicEditorPageAccessValidationHandlerTests(test_utils.EmailTestBase):
+    """Checks the access to the topic editor page and its rendering."""
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.signup(self.CURRICULUM_ADMIN_EMAIL, self.CURRICULUM_ADMIN_USERNAME)
+        self.add_user_role(
+            self.CURRICULUM_ADMIN_USERNAME, feconf.ROLE_ID_CURRICULUM_ADMIN)
+
+        self.topic_id = topic_fetchers.get_new_topic_id()
+        self.admin_id = self.get_user_id_from_email(self.CURRICULUM_ADMIN_EMAIL)
+        self.skill_id_2 = skill_services.get_new_skill_id()
+        self.skill_id = skill_services.get_new_skill_id()
+
+    def test_access_topic_editor_page_without_logging_in(self) -> None:
+        self.get_json(
+            '%s/can_access_topic_editor/%s' %(
+            ACCESS_VALIDATION_HANDLER_PREFIX,
+            self.topic_id), expected_status_int=401)
+
+    def test_access_topic_editor_page_with_guest_user(self) -> None:
+        self.signup(self.VIEWER_EMAIL, self.VIEWER_USERNAME)
+        self.login(self.VIEWER_EMAIL)
+        self.get_json(
+            '%s/can_access_topic_editor/%s' %(
+            ACCESS_VALIDATION_HANDLER_PREFIX,
+            self.topic_id), expected_status_int=401)
+        self.logout()
+
+        self.login(self.NEW_USER_EMAIL)
+        self.get_json(
+            '%s/can_access_topic_editor/%s' % (
+                ACCESS_VALIDATION_HANDLER_PREFIX, self.topic_id),
+            expected_status_int=401)
+        self.logout()
+
+    def test_access_topic_editor_page_with_curriculum_admin(
+        self
+    ) -> None:
+        self.login(self.CURRICULUM_ADMIN_EMAIL)
+        self.get_html_response(
+            '%s/can_access_topic_editor/%s' % (
+                ACCESS_VALIDATION_HANDLER_PREFIX, self.topic_id))
+        self.logout()
+
+    def test_cannot_access_topic_editor_page_with_invalid_topic_id(
+        self
+    ) -> None:
+        self.login(self.NEW_USER_EMAIL)
+        self.get_html_response(
```

Comment:
Ditto

Reply:
@DubeySandeep  Very Sorry for the late response,
The reason for using `get_html_response` here  is that the test expects an HTML response, using `get_json` leads to an assertion error (vice-versa is also true.) something like this
```
Traceback (most recent call last):
  File "/Users/harshkeshari/opensource/oppia/core/controllers/access_validators_test.py", line 551, in test_cannot_access_topic_editor_page_with_invalid_topic_id
    self.get_json(
  File "/Users/harshkeshari/opensource/oppia/core/tests/test_utils.py", line 2985, in get_json
    return self._parse_json_response(
  File "/Users/harshkeshari/opensource/oppia/core/tests/test_utils.py", line 2937, in _parse_json_response
    self.assertEqual(json_response.content_type, 'application/json')
AssertionError: 'text/html' != 'application/json'
- text/html
+ application/json
```